apis/v1alpha2/httproute_types.go

```diff
@@ -366,21 +367,21 @@ type HTTPQueryParamMatch struct {
 	// +kubebuilder:default=Exact
 	Type *QueryParamMatchType `json:"type,omitempty"`
 
-	// Values is a map of HTTP query parameters to be matched. It MUST contain
-	// at least one entry.
-	//
-	// The query parameter name to match is the map key, and the value of the
-	// query parameter is the map value.
+	// Name is the name of the HTTP query param to be matched. Name matching
+	// matching MUST be case-sensitive. (See
+	// https://tools.ietf.org/html/rfc7230#section-2.7.3).
 	//
-	// Multiple match values are ANDed together, meaning, a request must match
-	// all the specified query parameters to select the route.
-	//
-	// HTTP query parameter matching MUST be case-sensitive for both keys and
-	// values. (See https://tools.ietf.org/html/rfc7230#section-2.7.3).
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	Name string `json:"Name"`
+
+	// Value is the value of HTTP Header to be matched. Value matching
+	// matching MUST be case-sensitive. (See
```

Comment:
Can we just say the value matches if it is the same (string). Calling out case-sensitive leads people to wonder about what other special cases there are (there aren't any).

Reply:
The rationale for calling out case-sensitive here is that we're calling out for related fields as case-insensitive. I thought it would be helpful to contrast the case sensitivity for value with the case insensitivity for name. Of course if we didn't have the case-insensitive comment above I don't think this would be worthwhile.

Reply:
🤦 Just realized I have a typo from copy-pasting here. Will fix. Now that I realize this is for query param where both are case-sensitive, maybe it doesn't matter quite as much. I think the contrast with the very similar header matching text still may make it worthwhile, but don't have to stick with that.

Reply:
Fixed typo and removed case-sensitive statement since it is not present for header value either.